### ISSUE TITLE
Add validation to project edit update form

### ISSUE THREAD

**Issue by spwoodcock** [hotosm/fmtm#1308]
**Describe the bug**

- There is currently no validation.
- Clicking update calls the API and returns an error if no file is selected.

**Expected behavior**

- Validation must be provided for both the file upload and the category selection.

**Screenshots**

![image](https://github.com/hotosm/fmtm/assets/78538841/77b0075c-2491-43a0-a6d8-13768a43a936)

@manjitapandey

**Comment by manjitapandey** [comment]
@spwoodcock , We have removed updating of form category since it may affect the task splitting algorithm causing overall implementation to be disturbed. 
Do you suggest  allowing users to edit form category also?

**Comment by spwoodcock** [comment]
The XForm selection doesn't have an impact on the task splitting. It's the data extract that impacts that. Currently they are strongly intertwined during project creation, but they are actually different.

The XForm is just the questionnaire for ODK basically, so we should be able to update that afterwards, either with a custom form or a predefined one.

If the user uploads a custom form, they must provide a category too ideally.

**Comment by spwoodcock** [comment]
You raise a good point that if the user changed from say a buildings form to a highways form, then this would affect the data extract and splits. 

But users should not be re-using the same project to map different features & should create a new project instead.

My use case was if the user chose the wrong form initially (a mistake), but wanted to update it without creating a new project.

**Comment by nrjadkry** [comment]
Shouldn't the task splits be generated according to the number of features in it? Let's say, we split the tasks according to the number of buildings in it, but the users wants to map the educational institution, then this will generate splits with uneven number of features.

**Comment by spwoodcock** [comment]
You are right, but that is assuming the user is using the predefined xforms.

This all gets overly complicated when we need to support custom xforms and custom extracts.

They should be able to update with a custom xform regardless of the category selection. 

Perhaps we should prevent the user changing to a different predefined xform?

I think we need the option to:
- Upload a custom xform whenever. 
- Return back to the predefined xform for the category they have selected.

(with this in mind, during project creation we should probably set the category name to 'custom' if the user uploads a custom extract.)

**Comment by manjitapandey** [comment]
Also needs the validation on form, For now, any random XLSX file can be uploaded.

**Comment by manjitapandey** [comment]
The validation is done now but the error message needs to be modified. Currently, i encounteres 4 lines of error message which is not user friendly. 
![image](https://github.com/hotosm/fmtm/assets/97273021/74a7e0ec-2803-4709-bc85-0770eeb24ea2)

needs short and user friendly error message.

**Comment by manjitapandey** [comment]
@Sujanadh , @NSUWAL123 Can you share some initial finds and any solution we can move ahead with? 
@spwoodcock . Since the validation has been done and only issue is user facing error message, How can we provide user with user friendly error message.
